services/osparc-gateway-server/src/osparc_gateway_server/backend/utils.py

```diff
@@ -344,7 +345,12 @@ async def get_cluster_information(docker_client: Docker) -> ClusterInformation:
 async def get_next_empty_node_hostname(
     docker_client: Docker, cluster: Cluster
 ) -> Hostname:
-    cluster_nodes = await docker_client.nodes.list()
+    if not hasattr(get_next_empty_node_hostname, "counter"):
+        get_next_empty_node_hostname.counter = 0
```

Comment:
THOUGHT: does the variable set in the callable ``get_next_empty_node_hostname.counter`` acts as a global? i.e. keeps track of how many times the function is actually called? if so, would't it be more clear to have some kind of decorator that "counts function calls"?

Reply:
it's a bit of a static variable in C++ but in python.